sfa_dash/blueprints/admin.py

```diff
@@ -586,26 +626,31 @@ def post(self):
         try:
             permission_id = permissions.post_metadata(permission)
         except DataRequestException as e:
+            # If permission creation failed, retain form input and reload the
+            # page.
+            self.set_template_args()
             return render_template(
-                self.template, form_data=form_data, errors=e.errors)
+                self.template,
+                form_data=form_data,
+                errors=e.errors,
+                **self.template_args)
+
+        # Get the list of object uuids from the form
         objects = filter_form_fields('objects-list-', form_data)
+
         errors = {}
-        messages = {}
         for object_id in objects:
             try:
                 permissions.add_object(permission_id, object_id)
             except DataRequestException:
+                # collect errors about permissions that could not be added
                 errors[object_id] = [
                     f'Failed to add {self.data_type} to permission.']
-            else:
-                messages[object_id] = [
-                    f'Added {self.data_type} to permission.']
-        if errors:
-            return render_template(
-                self.template, errors=errors, messages=messages)
-        messages = {'Success': 'Permission created.'}
-        return redirect(url_for('admin.permissions',
-                                messages=messages))
+
+        # display any errors and redirect to the newly created permission.
+        self.flash_api_errors(errors)
+        flash('Success: Permission created.')
```

Comment:
can we simultaneously have errors and success?

Reply:
Yes. So in reviewing this I realized the current behavior is really confusing and obfuscates what's actually happening.

Current behavior:
1. Try to create a permission, if it fails, show the form with errors.
2. If the permission was created successfully, try to add the selected objects to the permission. Collect error and success messages. If errors occurred, show the form and print all of the messages. This is deceptive because we created a permission, but we're still on the "create a new permission" form. The user has no indication that they can access the permission that was created and will probably just try to create another one.

Behavior introduced in this PR:
1. Same as above.
2. Try to add the objects to the new permission, collecting an error for any failure. Then redirect the user to the new permission page with error messages(if they exist). The user will be able to see the new permission metadata and the objects that were successfully added along with error messages about those that failed. 

Other than these bugs, there are only two conditions in which users should see these errors:
-They don't have create permissions. They should see a failure at step 1 above.
- They have a `create permissions` permission, but not `update all permissions`. Then, they could create the permission buy adding objects would fail. This might be confusing at first, but now that they land on the created permission page they will see a list of errors and won't see the "add objects to permission" button.

Reply:
thanks that makes sense!